DOCS-2485: Add image and sensor data capture quickstart (#3074)
viamrobotics/docs: docs/get-started/quickstarts/collect-data.md

---
title: "Collect images or sensor data in 2 minutes"
linkTitle: "Collect data in 2 minutes"
type: "docs"
weight: 20
cost: 75
images: ["/platform/data.svg"]
description: "Use Viam to gather images or sensor data from your machine."
---

> Follow this guide to start collecting images or sensor data from your machine using the [data management service](/services/data/):

Author: My thought was that people might read "this guide" and think we mean the linked data page instead of the page they're already on, but that's probably not evident anyway so I'll apply your suggestion!

## Requirements

- A Linux or macOS computer:
  - A single-board computer (SBC) such as a Raspberry Pi, OR
  - A laptop or desktop
- A camera or a sensor.

>   Options:

Author: I think my original "Options" feels sorta short and blunt, but I also am not sure about "Options include" because I think these are all of the options, not just some of them, so I think I'll leave as-is for now

  - The webcam built into your computer

>   - A USB webcam plugged into your computer or SBC

Author: Okay! I'll change "connected" and then also change "into" to "to" to work with it.
```suggestion
  - A USB webcam connected to your computer or SBC
```

{{< /expand >}}
{{%expand "Step 2: Install viam-server" %}}

Navigate to the **CONFIGURE** tab of your machine's page in the [Viam app](https://app.viam.com).
Follow the {{< glossary_tooltip term_id="setup" text="setup instructions" >}} that appear on your new machine's **CONFIGURE** page to install `viam-server` on your computer and connect it to the Viam app.

![The Viam app DATA page showing sensor data from an air quality sensor.](/get-started/quickstarts/collect-data/setup-button.png)

{{% /expand%}}
{{< expand "Step 3: Configure a camera" >}}

{{<gif webm_src="/how-tos/configure-webcam.webm" mp4_src="/how-tos/configure-webcam.mp4" alt="The process described below." max-width="550px" class=aligncenter >}}

1. From the **CONFIGURE** tab of your machine's page in [the Viam app](https://app.viam.com/), click the **+** icon next to your machine part in the left-hand menu and select **Component**.

2. Select the `camera` type, then select the `webcam` model.

3. Enter a name or use the suggested name for your camera and click **Create**.

4. Click the **Save** button in the top right corner of the page to save your config.

> 5. Go to the **CONTROL** tab and expand the camera's remote control card to test the camera stream.

Author: To my ear this reads more awkwardly. Is it okay if I leave as-is?

   If you don't see an image stream, you need to [configure the `video_path` attribute](/components/camera/webcam/#using-video_path).

For more detailed information, including optional attribute configuration, see the [`webcam` docs](/components/camera/webcam/).

{{< /expand >}}
{{< expand "Step 4: Configure data capture on the camera" >}}

> {{<gif webm_src="/how-tos/capture-images.webm" mp4_src="/how-tos/capture-images.mp4" alt="The process described below." max-width="600px" class=aligncenter >}}

Author: Ooh very nice catch. It oddly doesn't seem to matter, but it wasn't intentional! I did this the other day too oops--fixing the only other place this appears in the docs: https://github.com/viamrobotics/docs/pull/3092

1. Return to the **CONFIGURE** tab of your machine's page in [the Viam app](https://app.viam.com/).
   Locate the configuration card for your camera.

2. Click the **Add method** button.

>    When the **Create data management service** prompt pops up, click to add the service to your machine.

Author: That's a good less colloquial suggestion ty!

   For example, if you have a Sensirion SHT3x-DIS temperature and humidity sensor, you should use the [`sensirion-sht3xd`](https://docs.viam.com/components/sensor/sensirion-sht3xd/) model of sensor.

   Once you determine which model to use, add it to your machine's configuration:

1. From the **CONFIGURE** tab of your machine's page in [the Viam app](https://app.viam.com/), click the **+** icon next to your machine part in the left-hand menu and select **Component**.

1. Select the `sensor` type, then select the model that matches your hardware.

   {{<imgproc src="/get-started/quickstarts/collect-data/config-sensor.png" resize="x1100" declaredimensions=true alt="The dropdown showing all sensor models." style="max-width:550px" >}}

1. Enter a name or use the suggested name for your sensor and click **Create**.

1. Add required attributes, such as information about how the sensor is connected to the board.
   You can find information on these attributes by clicking the name of your sensor model in the [supported models list](/components/sensor/#supported-models).

> 1. Click the **Save** button in the top right corner of the page to save your config.

Author: I just searched the docs and it seems that upper right is more common than top right, and that non-hyphenated is more common than not for both. Your suggestion is in line with some style guides such as Microsoft's https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/upper-left-upper-right but we don't use this and our `AvoidWithSubstitution.yml` contains the following lines:   `top left: upper left
  top right: upper right` so I shall adhere to that! Thank you for calling my attention to this!
```suggestion
1. Click the **Save** button in the upper right corner of the page to save your configuration.
```

   {{<imgproc src="/get-started/quickstarts/collect-data/get-readings-control-tab.png" resize="x1100" declaredimensions=true alt="The sensor card on the Control tab with some air quality readings." style="max-width:600px" >}}

{{% alert title="Important" color="note" %}}
If your sensor uses I<sup>2</sup>C, SPI, or serial port communication, you need to enable that type of communication in your board's settings.
For example, if you are using a Raspberry Pi, SSH to it and [enable serial communication in `raspi-config`](/get-started/prepare/rpi-setup/#enable-communication-protocols).
{{% /alert %}}

{{< /expand >}}
{{< expand "Step 5: Configure data capture on the sensor" >}}

{{<gif webm_src="/how-tos/capture-sensor-readings.webm" mp4_src="/how-tos/capture-sensor-readings.mp4" alt="The process described below." max-width="600px" class=aligncenter >}}

1. Return to the **CONFIGURE** tab of your machine's page in [the Viam app](https://app.viam.com/).
   Locate the configuration card for your sensor.

2. Click the **Add method** button.
   When the **Create data management service** prompt pops up, click to add the service to your machine.
   Leave the default settings.

> 3. Scroll back up to your sensor config card.

Author: Okay! My personal take is, we use both pretty interchangeably in the docs (similar-ish numbers of instances), as well as in conversation, and the shortened version is still quite similar but shorter, so *maybe* it's good for users to be exposed to both casually so it becomes part of their vocabulary? I don't feel strongly or mind changing this one just sharing thoughts.
